### ISSUE TITLE
MusicXML: full measure rests export fails

### ISSUE THREAD

**Issue by FelippeRoza** [frescobaldi/python-ly#98]
I spend the weekend working in the multi-measure rest feature in order to make it able to deal with \compressFullBarRests, but something was going wrong. Then I realized that the current method to deal with multi-measure rest fails with some notation.

The following ly code is compiled with Frescobaldi generating a 2 bar sheet:
`{
  R1  R1
}`

But it exports to MusicXML a single measure with two rests:

```xml
  <part-list>
    <score-part id="P1">
      <part-name />
    </score-part>
  </part-list>
  <part id="P1">
    <measure number="1">
      <attributes>
        <divisions>1</divisions>
        <time symbol="common">
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <note>
        <rest />
        <duration>4</duration>
        <voice>1</voice>
      </note>
      <note>
        <rest />
        <duration>4</duration>
        <voice>1</voice>
      </note>
    </measure>
  </part>
</score-partwise>
```

**Comment by FelippeRoza** [comment]
Apparently it works sometimes when bar division is explicitly declared with the symbol '|'. The lilypond compiler is somehow 'smarter' and is able to distribute the rests and notes to the appropriate measure without the bar symbols.

**Comment by uliska** [comment]
This is a correct interpretation, and having to specify barchecks (the `|`) is one known limitation.

The problem for Frescobaldi/python-ly is that when you have any music expression it is hard to know where it will finally end up in a score. When LilyPond parses an expression it definitely knows that.

However, python-ly *does* have support for calculating lengths (notice the "Pos:" entry in the editor's status bar?), and figuring out if it is possible to make python-ly's exporter smarter in this regard would also be a valuable part of your project.

**Comment by FelippeRoza** [comment]
Right. I will try to figure out how Frescobaldi calculates the position of the element and how this can be used to solve this problem.

**Comment by FelippeRoza** [comment]
This one appears to be fixed already. On current master, `{ R1 R1 }` exports two measures with one whole rest each, rather than one measure with two rests:

```xml
<measure number="1">
  <attributes>...</attributes>
  <note><rest /><duration>4</duration><voice>1</voice></note>
</measure>
<measure number="2">
  <note><rest /><duration>4</duration><voice>1</voice></note>
</measure>
```

Walking master's first-parent history, the measure count for that input goes 1 -> 3 -> 2, settling at 2 in 8faa97a (Merge PR #115, 2017-09-22); PR #112 changed where measures get created and #115 followed it. It has been 2 ever since.

Closing as fixed. Happy to reopen if there is a variant that still misbehaves.
